gateway/ln-gateway/src/lightning/ldk.rs

```diff
@@ -321,20 +347,24 @@ impl ILnRpcClient for GatewayLdkClient {
                             ..
                         } = payment_details.kind
                         {
-                            return Ok(PayInvoiceResponse {
+                            break Ok(PayInvoiceResponse {
                                 preimage: Preimage(preimage.0),
                             });
                         }
                     }
                     PaymentStatus::Failed => {
-                        return Err(LightningRpcError::FailedPayment {
+                        break Err(LightningRpcError::FailedPayment {
                             failure_reason: "LDK payment failed".to_string(),
                         });
                     }
                 }
             }
             fedimint_core::runtime::sleep(Duration::from_millis(100)).await;
-        }
+        };
+
+        drop(payment_lock_guard);
```

Comment:
nit: You don't need to explicitly drop it here, it will drop when it falls out of scope.

Reply:
Fixed.